File: dart/42.dart
class Solution {
    int min(int a, int b){
        return (a < b) ? a : b;
    }
    int trap(List<int> h) {
        int n = h.length, tw = 0, j = -1;
        List<int> st = [];
        for(int i = 0; i < n; i++)
            st.add(0);
        for (int i = 0; i < n; i++) {
            while (j > 0 && h[st[j]] < h[i]) {
                if (h[st[j - 1]] >= h[st[j]])
                    tw += (min(h[st[0]], h[i]) - h[st[j]]) * (st[j] - st[j - 1]);
                j--;
            }
            if (j != -1 && h[st[j]] <= h[i])
                st[j] = i;
            else
                st[++j] = i;
        }
        return tw;
    }
}

File: python3/42.py
class Solution:
    def trap(self, h: List[int]) -> int:
        n = len(h)
        tw = 0
        j = -1
        st = [0] * n
        for i in range(n):
            while j > 0 and h[st[j]] < h[i]:
                if h[st[j - 1]] >= h[st[j]]:
                    tw += (min(h[st[0]], h[i]) - h[st[j]]) * (st[j] - st[j - 1])
                j -= 1
            if j != -1 and h[st[j]] <= h[i]:
                st[j] = i
            else:
                j += 1
                st[j] = i
        return tw
